src/Configuration/Config.Abstractions/src/IConfigurationProvider.cs

```diff
@@ -37,6 +37,12 @@ public interface IConfigurationProvider
         /// </summary>
         void Load();
 
+        /// <summary>
+        /// Use the following values as the configuration data for this provider.
+        /// </summary>
+        /// <param name="data">The configuration data this provider should use.</param>
+        void Use(IDictionary<string, string> data);
```

Comment:
This name kind of sucks, considered a Load() overload but that seemed confusing too, open to other ideas here...

Reply:
- AddData
- UseData
- SetData

Reply:
`UseData` is my preference out of the suggestions, but I don't feel strongly about it.